experimental/generation/generator/packages/library/src/substitutions.ts

```diff
@@ -111,20 +123,19 @@ function substitutions(path: string, bindings: any, copies?: number, seed?: stri
             if (line.trim() === '') {
                 result.push(line)
             } else {
-                let generated = new Set<string>()
-                let all = line.match(/\${[^}*]+\*}/g) || []
-                for (let [key, variable] of state) {
+                const all = line.match(/\${[^}*]+\*}/g) || []
+                for (const [key, variable] of state) {
                     variable.index = all.includes(`\${${key}*}`) ? 0 : -1
                 }
                 do {
                     let missing = false
-                    for (let i = 0; i < copies; ++i) {
+                    for (var i = 0; i < copies; ++i) {
```

Comment:
Why `var` instead of `let`? I don't look like you are leveraging a hoisted declaration.